### DISCUSSION TITLE
Implementation of polar decomposition, sqrt for rank two tensor

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#24388] [Q&A General] [answered]
After poking around a bit, I'm happy to see there is already a method that does polar decomposition of a rank two tensor (in tensor mechanics [here](https://github.com/idaholab/moose/blob/0dc615af26b00e885ecc41fb034323d27bf63a5b/modules/tensor_mechanics/src/materials/lagrangian/ComputeLagrangianObjectiveStress.C#L190-L205)).

I know that a lot of the heavy lifting when doing the polar decomposition is in taking the square root of the tensor. I'd like to know how this is implemented in MOOSE. Looks like that polar decomposition method is calling `MathUtils::sqrt()`, but I'm having trouble finding where that is defined. Can anyone point me in the right direction?

**Comment by hugary1995** [comment, marked as answer]
That (and a few other common unary operators) is defined in `FactorizedRankTwoTensor.h`. It's just an eigenvalue decomposition behind the scenes.
